### ISSUE TITLE
Nullability constraint warnings are not reported for method signatures

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#32539]
No warnings reported for `C<object?>` in method signatures:
```C#
class C<T> where T : class
{
}

class Program
{
    static C<object?> F1() => throw null;

    static void F2(C<object?> c) { }
}
```

**Comment by jaredpar** [comment]
Dupe of #32953
